### DISCUSSION TITLE
Question: From New CNCF Contributor

### DISCUSSION THREAD

**Discussion by danielkurniadi** [crossplane/crossplane#4868] [General]
Hi team. I'm a (senior) software engineer with 4 years of experience in both big tech (TikTok) and startup (Gojek, Zendesk, Arta Finance). I'm mostly working with Golang / Python3 / Typescript as Backend / Devops / Data engineer.

I want to get up-to speed to help this project take off. However, I need more context so that my time and contribution can be maximal without giving spamming the main contributors here too much in my early ramp-up stage.

### Resources to Learn
Gathering from what I found. Please add more if you find anything useful to read

- [Slack](https://kubernetes.slack.com/?redir=%2Fmessages%2FCHD49TLE7) (#crossplane channel)
- [Documentation](https://docs.crossplane.io/) (books and tutorial)
- [Youtube](https://www.youtube.com/channel/UC19FgzMBMqBro361HbE46Fw) channelv=VFbIHc3NbJo&list=PL2h5ikWC8viKmhbXHo1epPelGdCkVlF16&pp=iAQB)
- [Github](https://github.com/crossplane/crossplane) (this code base)

ETA: 1-2 weeks of tutorials to ramp up.

### Project Cycle, Sprint and Milestones

I'm yet to find any other than Github issues but perhaps I can start there. If we happen to have external platform for managing sprints / feature roadmap, can someone in here help point out so that I can align with the direction of the team? Reason I asked is because:
- I want to try pick easier task in my 1st few weeks.
- Once I get advanced, I want to try having a more vertical or end-end responsibility for one or two specific features.
- I want to make sure that when the team give me the task, I can complete it within the expected time so that people effort is not wasted and I don't create blockers.
- Link or invite to sprint ticket platform will be nice to facilitate communication and progress update.

Will be great to know the challenge in this roadmap.

### Mentor or Maintainers 

Good to have someone in contact that is either part-time or full-time working on this project.

### Commitment

I will have 20-30 hours per week to commit to this project by early November - December 2023. I probably have more bandwidth in 2024 Q1-Q2. I'm also interested in exploring bare-metal clouds for enterprise solution for my own company.
```[tasklist]
### Tasks
```

**Comment by danielkurniadi** [comment]
If you want to book meeting with me:
- https://calendly.com/dkurniadi-daniel/open-source-1-1

**Comment by negz** [comment]
Hi @danielkurniadi! Thanks for offering to contribute.

Some other resources:
* https://github.com/crossplane/crossplane/tree/master/design - Design docs
* https://github.com/crossplane/crossplane/#special-interest-groups-sig - Specific feature focus areas

Are there any SIGs that sound particularly relevant to your interests? Here's a little more detail on some of the more active ones:

* [#sig-composition-environments](https://crossplane.slack.com/archives/C05BP6QFLUW) - Allow folks to load extra configuration from "the environment" (i.e. additional Kubernetes resources) when configuring how a composite resource should be created.
* [#sig-composition-functions](https://crossplane.slack.com/archives/C031Y29CSAE) - Allow folks to use a pipeline of (serverless style) Functions to configure how a composite resource should be created.
* [#sig-deletion-ordering](https://crossplane.slack.com/archives/C05BP8W5ALW) - Make sure Crossplane doesn't get deadlocked when deleting stuff.
* [#sig-devex](https://crossplane.slack.com/archives/C05U1LLM3B2) - Very broadly, improve the developer experience (CLI tools, better errors, faster feedback, etc).
* [#sig-e2e-testing](https://crossplane.slack.com/archives/C05C8CCTVNV) - Improve our (relatively new) E2E tests.

If you're unopinionated, I might suggest Functions. We could use folks who know TypeScript and Python to help us improve the experience of authoring new Composition Functions in those languages.

**Comment by danielkurniadi** [comment]
Thank you, @negz this is awesome. I'll start with the open issues and asking some of the sigs if there is anything I can help.

**Reply by negz** [reply]
Nice! I've moved this issue over to a GitHub discussion. I don't see many GitHub notifications so feel free to ping me on Slack if you get stuck with anything.
